cabal-install/src/Distribution/Client/Manpage.hs

```diff
@@ -78,9 +78,11 @@ manpageCmd pname commands flags
 
         unless (ec1 == ExitSuccess) $ exitWith ec1
 
-        pager <- fromMaybe "less" <$> lookupEnv "PAGER"
-        -- 'less' is borked with color sequences otherwise
-        let pagerArgs = if takeFileName pager == "less" then ["-R"] else []
+        pagerAndArgs <- fromMaybe "less -R" <$> lookupEnv "PAGER"
+        -- 'less' is borked with color sequences otherwise, hence -R
+        (pager : pagerArgs) <- case words pagerAndArgs of
```

Comment:
I am suprised this does not earn you a pattern completeness warning by GHC (or maybe it uses MonadFail).
Anyhow, the best practice is:
```haskell
(x, xs) <- case mylist of
  [] -> error-out
  x:xs -> pure (x, xs)
```

Reply:
Tanks! Updated.